hash/hasher_test.go

```diff
@@ -122,30 +137,97 @@ func TestComparatorBcryptFail(t *testing.T) {
 		mkpw(t, 72),
 	} {
 		t.Run(fmt.Sprintf("case=%d", k), func(t *testing.T) {
+			_, reg := internal.NewFastRegistryWithMocks(t)
 			mod := make([]byte, len(pw))
 			copy(mod, pw)
 			mod[len(pw)-1] = ^pw[len(pw)-1]
 
-			err := hash.CompareBcrypt(context.Background(), pw, mod)
+			err := hash.CompareBcrypt(context.Background(), reg.Config(context.Background()), pw, string(mod))
+			assert.Error(t, err)
+		})
+	}
```

Comment:
Can you please add test cases for the "legacy" bcrypt format we had until now?

Reply:
@zepatrik: could you explain what you want to test? Current approach does not allow "legacy" hash format and adding migration to the PR should convert "legacy" BCrypt hashes to "new" format with prefix. I could support "legacy" format if that desirable but then we would have additional complexity.

Reply:
Done in 584fbbf